demos/tutorials/Deconvolution1D.py

```diff
@@ -0,0 +1,164 @@
+"""
+One-dimensional deconvolution
+=============================
+
+    This tutorial walks through the process of solving a simple deconvolution problem
+    in a Bayesian setting.
+
+"""
+# %%
+# Setup
+# -----
+# We start by importing the necessary modules
+
+import cuqi
+import numpy as np
+import matplotlib.pyplot as plt
+
+# %%
+# Setting up the forward model
+# ----------------------------
+# We start by defining the forward model. In this case, we will use a simple
+# convolution model. The forward model is defined by the following equation:
+#
+# .. math::
+#    \mathbf{y} = \mathbf{A} \mathbf{x}
+#
+# where :math:`\mathbf{y}` is the data, :math:`\mathbf{A}` is the convolution (forward model)
+# operator, and :math:`\mathbf{x}` is the solution.
+#
+# The easiest way to define the forward model is to use the testproblem module.
+# This module contains a number of pre-defined test problems that contain the
+# forward model and synthetic data. In this case, we will use the
+# :class:`cuqi.testproblem.Deconvolution1D` test problem. We extract the forward model
+# and synthetic data from the test problem by calling the :func:`get_components` static
+# method.
+
+# Forward model and data
+A, y_data, info = cuqi.testproblem.Deconvolution1D.get_components(use_legacy=False)
```

Comment:
Maybe for a different PR, but I find the static method has confused me on multiple occasions. Might be just me, but 
`cuqi.testproblem.Deconvolution1D(...).get_components()`
with ... indicating some inputs, would be more natural. This runs, however inputs are ignored, as the static method uses the default parameters (or any given as input to the static method itself. Instead we currently require
`cuqi.testproblem.Deconvolution1D.get_components(...)`
If the above would work instead, we would also be able to create and save the object in a variable, say, prob, and then do prob.get_components().

Reply:
Agreed see #47